aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/entities/Entity.java

```diff
@@ -138,7 +138,10 @@ static String generateId() {
 
     /**
      * @return the subsegmentsLock
+     *
+     * @deprecated This is for internal use of the SDK and will be made private.
      */
+    @Deprecated
     ReentrantLock getSubsegmentsLock();
```

Comment:
I'm not sure how common of a use case, but what if a customer wants to perform an operation on a group of subsegments in a thread-safe manner? Should we keep this API just in case?